### ISSUE TITLE
GOPHER file X_allTracks.bed does not show all probes for unbalanced digests

### ISSUE THREAD

**Issue by hansenp** [TheJacksonLaboratory/Gopher#241]
If I create extended viewpoints for gene CYP1B1 (hg19) and allow unbalanced margins, the number of probes for unbalanced digests show two times b_min probes in total.

 ![image](https://user-images.githubusercontent.com/10495485/43474283-b4a26874-94f2-11e8-8e28-649c839acc17.png)

But if I open the URL in UCSC and load the file `<project_name>_allTracks.bed` (produced by GOPHER) as custom track the digest has not 2 times b_min probes.

![image](https://user-images.githubusercontent.com/10495485/43474387-075ad40c-94f3-11e8-9ecf-dfc33f541d56.png)

**Comment by hansenp** [comment]
I will check if the exported file for probes is also affected.

**Comment by hansenp** [comment]
I seems to be the case that the 2 times bmin probes of unbalanced digests do not have unique coordinates. Therefore, bmin probes are removed during export.

I'm working on this issue on branch iss241.

**Comment by hansenp** [comment]
Someone refactored the code for setting usable baits in class `Segment`. The functions `setUsableBaitsForUpstreamMargin` and `setUsableBaitsForDownstreamMargin` are intended to set all probes for the margin at once. Therefore, the lists `baitListUpStreamMargin` and `baitListDownStreamMargin` were cleared before selecting probes in the original version. This was changed, which caused the bug.

I changed this. Now the exported probe file contains the correct number of probes:

![image](https://user-images.githubusercontent.com/10495485/43513076-12db8b20-957d-11e8-98e8-c1a7a6e56a06.png)

@pnrobinson: I noticed that we are loosing lot's of time with issues like this. I would suggest to discuss this and define some guidelines for refactoring code written by others.
